track_test.go

```diff
@@ -53,3 +55,101 @@ func TestOnEnded(t *testing.T) {
 		}
 	})
 }
+
+type fakeRTCPReader struct {
+	mockReturn chan []byte
+	end        chan struct{}
+}
+
+func (mock *fakeRTCPReader) Read(buffer []byte, attributes interceptor.Attributes) (int, interceptor.Attributes, error) {
+	select {
+	case <-mock.end:
+		return 0, nil, io.EOF
+	case mockReturn := <-mock.mockReturn:
+		if len(buffer) < len(mock.mockReturn) {
+			return 0, nil, io.ErrShortBuffer
+		}
+
+		return copy(buffer, mockReturn), attributes, nil
+	}
+}
+
+type fakeKeyFrameController struct {
+	called chan struct{}
+}
+
+func (mock *fakeKeyFrameController) ForceKeyFrame() error {
+	mock.called <- struct{}{}
+	return nil
+}
```

Comment:
Not sure where to put this ? Do you want to push this to the end of the file since they are just test utils ?